feat(eslint-config): introducing shared eslint-config for esl projects (#2684)
exadel-inc/esl: eslint-config/README.md

# [ESL](../../../) Shared ESLint Configuration

Authors: *Anastasiya Lesun, Alexey Stsefanovich (ala'n)*.

<a name="intro"></a>

Packages maintained by ESL Team often use ESLint to ensure code quality and consistency. 
To simplify the process of configuring ESLint for these packages, we have developed a shared ESLint configuration. 
This configuration is designed to be used as a base for ESLint configuration in projects that use ESL package, 

> or desided to follow ESL source code style.

Reviewer: ```suggestion
or decide to follow ESL source code style.
```

<a name="installation"></a>

### Installation

To use ESL shared ESLint configuration, you need to install it as npm package:

```bash
npm install --save-dev @exadel/eslint-config-esl
```

> Ensure that you have the ESLint package of version 9.0.0 or higher, shared configuration distributed only as Flat ESLint config.

Reviewer: ```suggestion
Ensure that you have the ESLint package of version 9.0.0 or higher, shared configuration is distributed only as flat ESLint config.
```

Once installed, the configuration needs to be added in eslint configuration file:

```js
module.exports = [
    // ESLint configuration ...

    // Apply ESL Shared ESLint Configuration
    ...require('@exadel/eslint-config-esl').typescript,
    ...require('@exadel/eslint-config-esl').recommended,
];
```

<a name="configuration"></a>

### Configuration

There are no additional configuration options for ESL Shared ESLint Configuration at that moment. 

> Please be aware that we asre in progress of standardizing rule list and currently considering of suporrting EcmaScript in addition to TypeScript.

Reviewer: ```suggestion
Please be aware that we are in the process of standardizing rule list and are currently considering supporting EcmaScript in addition to TypeScript.
```

> Still yor are able to override any rule from the configuration in your project's ESLint configuration file 

Reviewer: ```suggestion
Still you are able to override any rule from the configuration in your project's ESLint configuration file 
```

by declaring it in the `rules` section of the configuration file after applying the shared configuration.

### Inner Plugins

ESL Shared ESLint Configuration includes several inner plugins that are used to provide additional rules and configurations for ESLint.

> Here is the list of included plugins and thair ESLint aliases:

Reviewer: ```suggestion
Here is the list of included plugins and their ESLint aliases:
```
